Revise telemetry_server - Add sys_status

bazfp opened the pull request:
- Made publish messages `SYNC`
- Removed unneeded rpcs from the `telemetry_server.proto` 
- Added support for more telemetry message types (`sys_status`, `extended_sys_state`)
- VTOL types
- RC receiver types
~Added Vehicle Arming~ Moved to `action_server` - https://github.com/mavlink/MAVSDK-Proto/pull/244
~Added Flight_modes~ Moved to `action_server` - https://github.com/mavlink/MAVSDK-Proto/pull/244

julianoes commented on protos/telemetry_server/telemetry_server.proto:
```diff
@@ -13,37 +13,48 @@ option java_outer_classname = "TelemetryServerProto";
     // Publish to 'unix epoch time' updates.
-    rpc PublishUnixEpochTime(PublishUnixEpochTimeRequest) returns(PublishUnixEpochTimeResponse) {}
+    rpc PublishUnixEpochTime(PublishUnixEpochTimeRequest) returns(PublishUnixEpochTimeResponse) { option (mavsdk.options.async_type) = SYNC; }
+    // Subscribe to ARM/DISARM commands
+    rpc SubscribeArmDisarm(SubscribeArmDisarmRequest) returns(stream ArmDisarmResponse) { option (mavsdk.options.async_type) = ASYNC; }
+    // Subscribe to DO_SET_MODE
+    rpc SubscribeDoSetMode(SubscribeDoSetModeRequest) returns(stream DoSetModeResponse) { option (mavsdk.options.async_type) = ASYNC; }
```
Oh, I would have expected this to be in an `ActionServer` plugin. Was that not the intention?

julianoes replied:
Alternatively, you could make an autopilot_server plugin and have it contain both.

julianoes commented on protos/telemetry_server/telemetry_server.proto:
```diff
@@ -489,6 +589,18 @@ message Imu {
+// Arming message type
+message ArmDisarm
+{
+    int32 arm = 1; // Should vehicle arm
+    int32 force = 2; // Should arm override pre-flight checks
```
Shouldn't these be booleans?

julianoes commented on protos/telemetry_server/telemetry_server.proto:
```diff
@@ -188,6 +242,35 @@ message PublishUnixEpochTimeResponse {
+message ArmDisarmResponse
+{
+    TelemetryServerResult telemetry_server_result = 1;
+    ArmDisarm arm = 2;
```
Why is this required as well when it already returns whether the arm or disarm command succeeded or not?

bazfp replied:
You cannot tell if it is an "Arm" or "Disarm" event

julianoes commented on protos/telemetry_server/telemetry_server.proto:
```diff
@@ -306,6 +396,16 @@ enum FlightMode {
 }
 
+// Maps to MAV_VTOL_STATE
+enum VTOLState
```
```suggestion
enum VtolState
```

julianoes reviewed (commented):
I'm a bit confused/concerned about the additional logic in the telemetry server which seems to be more around action rather than telemetry.

bazfp commented:
I've removed all `action_server` logic to it's own PR and removed some of the publishing which don't map to telemetry plugin (`Health` & `FlightModes` all depend on the heartbeat, which is not handled here)

`telem_server` looking leaner and cleaner.
